internal/security/fileprovider/tokenconfig.go

```diff
@@ -80,3 +87,89 @@ func LoadTokenConfig(fileOpener fileioperformer.FileIoPerformer, path string, to
 
 	return nil
 }
+
+// GetTokenConfigFromEnv function gets a list of token service keys
+// from environment variable and populates the default configuration with
+// default token parameters and policies
+// the function returns a TokenConfFile map instance and error if any
+// if the environment variable is not present or the value of that is empty
+// then it will return empty map
+// if the value for the list is not well-formed, not comma-separated
+// then it will return an error
+func GetTokenConfigFromEnv() (TokenConfFile, error) {
+	emptyTokenConfig := make(TokenConfFile)
+
+	addTokenList := os.Getenv(addSecretstoreTokensEnvKey)
+	if strings.TrimSpace(addTokenList) == "" {
+		return emptyTokenConfig, nil
+	}
+
+	// the list of service names is comma-separated
+	tokenConfigFromEnv := make(TokenConfFile)
+	serviceNameList := strings.Split(addTokenList, ",")
+	// regex for valid service name as key
+	// the service name eventually becomes part of the URL to Vault's API call
+	// Based upon the RFC 3986: https://tools.ietf.org/html/rfc3986#page-12,
+	// the following characters are reserved characters for URI's and thus NOT allowed:
+	// gen-delims  = ":" / "/" / "?" / "#" / "[" / "]" / "@"
+	// sub-delims  = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="
+	// backslash (\) also is not allowed due to being as a delimiter for URI directory in Windows
+	// and the regular alphanumeric characters like A to Z, a to z, 0 to 9, underscore (_),
+	// -, ~, %, ^,  {, }, |, <, >, . are allowed.
+	// and the the length of the name is upto 512 characters
+	serviceNameRegx := regexp.MustCompile(`^[\w. \~\%\^\-\|\<\>\{\}]{1,512}$`)
```

Comment:
@jim-wang-intel  Please remove "%"  as well, because that is used to encode the reserved characters and could be used to bypass the above filter -- for example, `%24100` would encode `$100`, bypassing the above filter.

Reply:
just saw this- good catch and changed.